src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs

```diff
@@ -192,7 +192,11 @@ private void SanitizeScheduledPollsLocked()
 
                 bool IsValid(VotingData currentScheduledData)
                 {
-                    // Remove scheduled voting data that can be found in pending polls.
+                    if (currentScheduledData.Key == VoteKey.AddFederationMember)
+                        // Only vote on pending polls that are not too old to vote on.
+                        return pendingPolls.Any(x => x.VotingData == currentScheduledData && !IsTooOldToVoteOn(x));
+
+                    // Remove scheduled voting data that relate to too-old pending polls.
```

Comment:
Why add the line above? Is the one below not going to cover it anyway? (Line 200)

Reply:
The current code does not have the ability to remove scheduled votes based on those votes not relating to pending polls - simply because that is not a requirement for other poll types. The other poll types can be created by the mere existence of a vote whilst "add member" polls cannot. "Add member" polls are only created when a voting request is witnessed.

Reply:
Both those LINQ statements check the pending polls collection and filters by `currentScheduledData` and `IsTooOldToVoteOn`. The one checks if its too old and returns false and the other checks if its NOT too old and returns true. Isnt that logic then one in the same?

Reply:
"The current code (line 200) does not have the ability to remove scheduled votes based on those votes not relating to pending polls". In contrast the new code does.